crates/nu-plugin/src/plugin/interface/plugin.rs

```diff
@@ -96,10 +97,28 @@ struct PluginCallState {
     ctrlc: Option<Arc<AtomicBool>>,
     /// Channel to receive context on to be used if needed
     context_rx: Option<mpsc::Receiver<Context>>,
+    /// Channel for plugin custom values that should be kept alive for the duration of the plugin
+    /// call. The plugin custom values on this channel are never read, we just hold on to it to keep
+    /// them in memory so they can be dropped at the end of the call. We hold the sender as well so
+    /// we can generate the CurrentCallState.
+    keep_plugin_custom_values: (
+        mpsc::Sender<PluginCustomValue>,
+        mpsc::Receiver<PluginCustomValue>,
+    ),
     /// Number of streams that still need to be read from the plugin call response
     remaining_streams_to_read: i32,
 }
 
+impl Drop for PluginCallState {
+    fn drop(&mut self) {
+        // Clear the keep custom values channel, so drop notifications can be sent
+        for value in self.keep_plugin_custom_values.1.try_iter() {
```

Comment:
We assume if there is some value around it is already waiting in the channel so we never block here?

Reply:
Yeah, we can't block here. We're also still holding the sender so it would wait forever.

If for some reason there were a race condition, anything left in the channel / sent after would also still be dropped. This `Drop` implementation is not strictly necessary for correctness, anyway - it's really just there for the debug logs.